### DISCUSSION TITLE
Bugs compilation from the Barotrauma Custom Game Community

### DISCUSSION THREAD

**Discussion by AziKarnage** [FakeFishGames/Barotrauma#14804] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

This is a rough list of things occasionally brought up by the CGC comprised of at least 800 people

baro bug report-

Vanilla sub-components are wrench-able when vanilla wiring lock is on

Shell pieces with an angle above 45 degrees became impossible to climb up despite being easy to do so prior, similar objects had the same effect - Blood In The Water Update lots of maps were broken including the compound /from Middle East project/ as well as /war on Europa, and Apocalyptic Nevada and Evergreens house raid/ which have not yet returned to use

Beacons were reworked in version 0.13.0.11, making the fuel rods invincible upon activation and junction boxes in later versions too. Although a good change for the campaign, this makes it difficult to make a game mode about surviving in a beacon station.

The grappling hook was broken due to characters being unable to grapple and rise on structures only lower, this broke most cave maps and with that some adventure games
/ I have heard in conversation it was taken out intentionally due to bugs with water however we considered this more of a feature and would like to see it return

once you connect to the server you need to first process every event that happened before joining and only then allowed to synchronize with the modern state of the game. But this process takes time, that is counted as the time to get synced with the rest of the lobby. So if the rounds been going for some time new players are 100% certain to time out

the p2p nature of baros multiplayer makes it so one player joining or having a bad connection means lags, desyncs and other net issues for the rest of the lobby
creating too much gibbs (for example if a powerful explosion gibs a bunch of people) it can just randomly desync and kick players

light-components states sometimes are not synchronized between players 

" spamming ragdoll makes you invincible to hitscans " unsure how accurate this is

Monsters are difficult to control having very jank desynced movement

Connection issues have progressively gotten worse 

"Dedicated servers cant pick up the slack"

Todd wants more hotkeys, we will accept Todds request being ignored

Job Icons are too large and you can only see the top vanilla 3 or 6 but many game modes require many more roles and the ability to change them in game which now requires rejoining and changing from lobby

Brokensprite doesnt effect wearables

roleplay mods that change names for players make it so admins have a harder time, an admin version of viewing the player lobby or vanilla side assistance if coordinated with those rp nerds (not me) might be helpful

inventory icons are hard coded

the settings to ensure people didnt lose skills after death were reported to not function in some servers /unverified 

Campaign mode sometimes doesn't show the option to respawn which breaks economy based game modes/ on the same note we request economy and campaign settings for sandbox and mission

Also pls make human ai no longer hardcoded, there is totally insane potential that comes from this community and the more code we can mess with the crazier it gets

Some is just copy pasted from what people said when prompted about any relevant bugs or changes needed for the custom games community

### Reproduction steps

The most important issues were the scaling structures update/blood in the water breaking maps. 

If you go to apocalyptic Nevada and try to climb the mountain or in Middle East Project and try scale some small boxes you will have significant trouble as compared to before where characters would glide over.

Joining any long host especially Middle East project or Combine Assault will show you lagging out for 15 seconds to several minutes when you join. The longer it's been going and the more explosions the worse it is and more likely to crash.

I do not know how to test light components being desynced as this was only reported one time

In pop modes spamming rag doll significantly decreases chances of being hit, again Middle East or maybe flintlock pvp are good types of servers to test this, it could also just be related to lag of the shooter however. Either way the tactic is quality, I don't really personally see a need to fix it unless it really is broken.

For monsters, use a veteran mud raptor and navigate convict station (while it is full of water) while trying to hit things, after a while the body gets twisted into itself and unable to hit people right in front of it, requiring a reset of the body via moving left or right and then pressing space as you switch back to the direction of your target. 

Job icons can be tested just by trying to make a new character in game and trying to select any roles other than the first 6, before it would just shrink all roles down for you to still see them but you'll see now its only ever the first 6 and maybe the outline of the three past it.

No information about the broken sprite issue 

the issue of role-play mods with names occurred on convict station where people spawned with rp names,  and their names were removed from the tab menu and crew list for rp and immersion. However there is no way to select crew after this or see who they are. If they type in chat we can click their name but the griefers who only use mic are unstoppable, and we can only select their name when they join not leave so they evade bans this way

Skill loss settings didn't work on Nevada, this is an uncommon issue reported 

When hosting m12s server with respawns enabled and in campaign mode, the menu would appear asking if you wish to respawn or not, to which some people misclicked and chose perms death. After they could not see any box or options to respawn

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Other

### -

Both dedicated and regular hosts of multiplayer

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by girlyguppy** [comment]
I especially like the beacon survival idea. in general, forcing anything like that because it works good in campaign is a no-no for custom games.
brokensprite not affecting wearables is hugely annoying for modding.
